### PR TITLE
Refactor the readme and add instructions to uninstall agents

### PR DESCRIPTION
**Pull request by digitalfishpond**
Change-Id: I79a9db608c233a867b0efb180eac7ff1722c6666

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,52 +1,25 @@
...
-This project allows to create a self-extracting installer for Monasca agents
-([metrics agent](https://github.com/openstack/monasca-agent), log agent and, in future, events agent).
-The aim of the installer is to easily install and configure agents on the target host.
+This project includes everything needed to create self-extracting installers 
+for Monasca agents as well as for the monasca-ui Horizon plugin, allowing the 
+user to easily install and configure agents on the target host.
 
-Additionally, the monasca-ui Horizon plugin is also supported.
+The agents currently supported are the [metrics agent](https://github.com/openstack/monasca-agent) 
+and the log agent. (There are plans to support an events agent in the future, ).
```
Comment (haruki-yamanashi) [inline README.md]: What's `,`  in the last?

Comment (digitalfishpond) [reply README.md]: done

**Comment by haruki-yamanashi** [comment]
Please add `--service_type` as argument for running Metric agent installer and parameter info.

File: README.md [hunk 2]
```diff
@@ -164,7 +153,45 @@ For more parameters, please see [Monasca Agent Documentation](https://github.com
 
 This will create and run a new service file `/etc/systemd/system/monasca-agent.service` with the configuration set as per the arguments mentioned above. 
 
-### Log agent
+### Uninstalling Metrics agent
+Stop the services:
```
Comment (haruki-yamanashi) [inline README.md:157]: Also execute `systemctl disable monasca-agent`

File: README.md [hunk 3]
```diff
@@ -164,7 +153,45 @@ For more parameters, please see [Monasca Agent Documentation](https://github.com
 
 This will create and run a new service file `/etc/systemd/system/monasca-agent.service` with the configuration set as per the arguments mentioned above. 
 
-### Log agent
+### Uninstalling Metrics agent
+Stop the services: 
+```
+systemctl stop monasca-agent
+```
+Delete created files:
```
Comment (haruki-yamanashi) [inline README.md:161]: Other files created by installer (and should be removed I think)
- /opt/monasca-agent
- /etc/sudoers.d/mon-agent

File: README.md [hunk 4]
```diff
@@ -164,7 +153,45 @@ For more parameters, please see [Monasca Agent Documentation](https://github.com
...
+sudo rm -rf /etc/monasca/agent/conf.d/disk.yaml
+sudo rm -rf /etc/monasca/agent/conf.d/kafka_consumer.yaml 
+sudo rm -rf /etc/monasca/agent/conf.d/load.yaml
+sudo rm -rf /etc/monasca/agent/conf.d/memory.yaml
+sudo rm -rf /etc/monasca/agent/conf.d/network.yaml
+sudo rm -rf /etc/monasca/agent/conf.d/postfix.yaml
+sudo rm -rf /etc/monasca/agent/conf.d/process.yaml
+sudo rm -rf /etc/monasca/agent/conf.d/zk.yaml
+```
+Finally, delete the service related to the agent.
```
Comment (haruki-yamanashi) [inline README.md:177]: Remove `mon-agent` user and execute `systemctl daemon-reload` after removing service related file.

File: README.md [hunk 5]
```diff
@@ -164,7 +153,45 @@ For more parameters, please see [Monasca Agent Documentation](https://github.com
...
 
-### Log agent
+### Uninstalling Metrics agent
+Stop the services: 
+```
+systemctl stop monasca-agent
+```
+Delete created files: 
+
+*(Note: there may be files in /etc/monasca/agent/ that are related to different agents, so only remove the ones listed.)*
```
Comment (haruki-yamanashi) [inline README.md:163]: Isn't it good to remove `/etc/monasca`?

File: README.md [hunk 6]
```diff
@@ -203,8 +230,37 @@ To include all the files in a directory, use the `*` wild card (eg. `/var/log/*`
...
+### Uninstalling Logs agent
+In the following description, 
+[target_dir] is the target directory specified at the time of running log-agent-[version].run
+
+stop the service:
+```
+systemctl stop monasca-log-agent
+```
+
+delete obsolete files:
```
Comment (haruki-yamanashi) [inline README.md:242]: I think `[target_dir]` means e.g. `/opt/monasca-log-agent` and under the directory, following files exist.

- bin
- conf
- logstash-[version]

So change as below.
`rm -r [target_dir]/bin`
`rm -r [target_dir]/conf`
`rm -r [target_dir]/monasca-log-agent`

BTW, isn't it good to remote `[target_dir]`?

Comment (haruki-yamanashi) [inline README.md:242]: `/etc/systemd/system/monasca-log-agent.service` is also created by Log agent installer (and should be removed I think).

File: README.md [hunk 7]
```diff
@@ -203,8 +230,37 @@ To include all the files in a directory, use the `*` wild card (eg. `/var/log/*`
 
 Additionally, you can add the `--no_service` to omit the step of automatically creating `monasca-log-agent.service` in `/etc/systemd/system/`
 
-### Monasca-ui
+### Uninstalling Logs agent
+In the following description, 
+[target_dir] is the target directory specified at the time of running log-agent-[version].run
+
+stop the service:
```
Comment (haruki-yamanashi) [inline README.md:237]: Please also disable service, `systemctl disable monasca-log-agent`

File: README.md [hunk 8]
```diff
@@ -255,3 +311,27 @@ After that, you need to restart the apache server:
...
+Remove the line with the instruction to append site-packages in the wsgi file:
+
+Open file: `<horizon_dir>/openstack_dashboard/wsgi/django.wsgi`, delete the line: `sys.path.append("<monasca_ui_dir>/lib/python2.7/site-packages/")` and save.
+
+Delete created files: 
+```
+rm -rf <installation_dir>/monasca_ui
+```
+
+Delete the installer:
```
Comment (haruki-yamanashi) [inline README.md:329]: Is it necessary to remove Horizon Plugin installer?

File: README.md [hunk 9]
```diff
@@ -203,8 +230,37 @@ To include all the files in a directory, use the `*` wild card (eg. `/var/log/*`
...
+```
+
+delete obsolete files: 
+```
+rm -f [target_dir]/conf
+rm -f [target_dir]/monasca-log-agent
+rm -f [target_dir]/logstash-[version]
+rm -f /etc/systemd/system/monasca-log-agent.service
+```
+
```
Comment (haruki-yamanashi) [inline README.md:249]: Please execute `systemctl daemon-reload` after removing files.

File: README.md [hunk 10]
```diff
@@ -203,8 +230,37 @@ To include all the files in a directory, use the `*` wild card (eg. `/var/log/*`
 
 Additionally, you can add the `--no_service` to omit the step of automatically creating `monasca-log-agent.service` in `/etc/systemd/system/`
 
-### Monasca-ui
+### Uninstalling Logs agent
```
Comment (haruki-yamanashi) [inline README.md:233]: After uninstalling process, `monasca-log-agent` service isn't removed.
Is this the other bug?

File: README.md (hunk 2 above)
Comment (matrixik) [reply README.md:157]: Done

File: README.md (hunk 3 above)
Comment (matrixik) [reply README.md:161]: Done

File: README.md (hunk 5 above)
Comment (matrixik) [reply README.md:163]: I changed it to removing `/etc/monasca/agent/`, we don't know if user have any other stuff from monasca installed there.

File: README.md (hunk 4 above)
Comment (matrixik) [reply README.md:177]: Done

File: README.md (hunk 7 above)
Comment (matrixik) [reply README.md:237]: Done

File: README.md (hunk 6 above)
Comment (matrixik) [reply README.md:242]: It's removed in line 247.

Comment (matrixik) [reply README.md:242]: Done

File: README.md (hunk 9 above)
Comment (matrixik) [reply README.md:249]: Done

File: README.md (hunk 8 above)
Comment (matrixik) [reply README.md:329]: Removed

File: README.md (hunk 10 above)
Comment (matrixik) [reply README.md:233]: Did you checked before or after running `systemctl daemon-reload`?

**Comment by matrixik** [comment]
Closing and replacing with #25

**Pull request closed without merging**